### PR TITLE
Feature/dswo 27 need to know and venue changes

### PR DESCRIPTION
**Pull request by Firdaussi**
Ticket URL: https://national-archives.atlassian.net/jira/software/projects/DSWO/boards/121?selectedIssue=DSWO-27

## About these changes

Additions to the Whats-On page: added venue directions (URL), need to know button text and URL. Modified venue space name to be non-mandatory.

## How to check these changes

In wagtail, What's On page creation, new fields as outlined above are now available to the user.

## Before assigning to reviewer, please make sure you have

- [Y] Checked things thoroughly before handing over to reviewer
- [Y] Checked PR title starts with ticket number as per project conventions to help us keep track of changes
- [Y] Ensured that PR includes only commits relevant to the ticket
- [x] Waited for all CI jobs to pass before requesting a review
- [x] Added/updated tests and documentation where relevant

## Merging PR guidance

Follow [docs\developer-guide\contributing.md](https://nationalarchives.github.io/ds-wagtail/developer-guide/contributing/)

## Deployment guidance

Follow [docs\infra\environments.md](https://nationalarchives.github.io/ds-wagtail/infra/environments/)

### CODE REVIEW

File: etna/whatson/migrations/0005_eventpage_need_to_know_button_link_and_more.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+                max_length=30,
+                verbose_name="need to know button text",
+            ),
+        ),
+        migrations.AddField(
+            model_name="eventpage",
+            name="venue_directions",
+            field=models.URLField(
+                blank=True,
+                help_text="A link to the venue’s ‘How to find us’ page.",
```
Comment (jamesbiggs) [inline etna/whatson/migrations/0005_eventpage_need_to_know_button_link_and_more.py:37]:
```suggestion
                help_text="A link to the venue's 'How to find us' page.",
```

I'm not sure if we should use `‘`

File: etna/whatson/models.py [hunk 2]
```diff
@@ -514,10 +529,19 @@ class EventPage(ArticleTagMixin, TopicalPageMixin, BasePageWithIntro):
...
         blank=True,
         help_text=_("The name of the venue space."),
     )
 
+    venue_directions = models.URLField(
+        max_length=255,
+        verbose_name=_("venue directions"),
+        null=True,
+        blank=True,
+        help_text=_("A link to the venue’s ‘How to find us’ page."),
```
Comment (jamesbiggs) [inline etna/whatson/models.py:542]:
```suggestion
        help_text=_("A link to the venue's 'How to find us' page."),
```

Same here

**Review by jamesbiggs — COMMENTED** [review]
Hi Ian, nice work, just some comments on it here 😄

**Comment by jamesbiggs** [comment]
Also, one missed thing, "`venue_space_name` should not be mandatory for Hybrid or In person events.", we can change this by editing the `clean()` function on lines `715-757`. If you remove the lines in the in person and hybrid conditions, just so it has `venue_address` in there, that'll do the trick 😄

File: etna/whatson/models.py (hunk 2 above)
Comment (Firdaussi) [reply etna/whatson/models.py:560]: Replaced the apostrophe's with standard one. Don't why this happened in the first place.

File: etna/whatson/models.py [hunk 3]
```diff
@@ -532,10 +547,19 @@ class EventPage(ArticleTagMixin, TopicalPageMixin, BasePageWithIntro):
     venue_space_name = models.CharField(
         max_length=255,
         verbose_name=_("venue space name"),
```
Comment (Firdaussi) [inline etna/whatson/models.py:549]: As per Ashley's remarks, ‘Venue space name’ should not be mandatory for Hybrid or In person events. But this is a specialised case so I have removed the nullability of this field and altered code in the clean() function below.

**Pull request merged**